README.md

```diff
@@ -1,51 +1,238 @@
+- Lookahead decoding that attempts to guess multiple tokens based on historical context and corrects this guess using LLMs. See [here](https://lmsys.org/blog/2023-11-21-lookahead-decoding/) for more detailed overview
+- KVCache token eviction algorithm that reduces size of the KVCache by pruning less impacting tokens.
+
+Additionally, OpenVINO™ GenAI library implements continuous batching approach to use OpenVINO within LLM serving. Continuous batching library could be used in LLM serving frameworks and supports following features:
+- Prefix caching that caches fragments of previous generation requests and corresponding KVCache entries internally and uses them in case of repeated query. See [here](https://google.com) for more detailed overview
+
+## Installing OpenVINO GenAI
+
+```sh
+    # Installing OpenVINO GenAI via pip
+    pip install openvino-genai
+
+    # Install optimum-intel to be able to download, convert and optimize LLMs from Hugging Face
+    # Optimum is not required to run models, only to convert and compress
+    pip install optimum[openvino]
+
+    # (Optional) Install (TBD) to be able to download models from Model Scope
+    #pip install optimum[openvino]
+```
+
+## Performing text generation 
+<details>
+For more examples check out our [LLM Cheat Sheet](https://docs.openvino.ai)
+
+### Converting and compressing text generation model from Hugging Face library
+
+```sh
+#(Basic) download and convert to OpenVINO TinyLlama-Chat-v1.0 model
+optimum-cli export openvino --model "TinyLlama/TinyLlama-1.1B-Chat-v1.0" --weight-format fp16 --trust-remote-code "TinyLlama-1.1B-Chat-v1.0"
+
+#(Recommended) download, convert to OpenVINO and compress to int4 TinyLlama-Chat-v1.0 model
+optimum-cli export openvino --model "TinyLlama/TinyLlama-1.1B-Chat-v1.0" --weight-format int4 --trust-remote-code "TinyLlama-1.1B-Chat-v1.0"
+```
+
+### Run generation using LLMPipeline API in Python
+
+```python
+import openvino_genai as ov_genai
+#Will run model on CPU, GPU or NPU are possible options
+pipe = ov_genai.LLMPipeline("./TinyLlama-1.1B-Chat-v1.0/", "CPU")
+print(pipe.generate("The Sun is yellow because", max_new_tokens=100))
+```
+
+### Run generation using LLM Pipeline in C++
+
+Code below requires installation of C++ compatible package (see [here](https://docs.openvino.ai/2024/get-started/install-openvino/install-openvino-genai.html#archive-installation) for more details)
+
+```cpp
+#include "openvino/genai/llm_pipeline.hpp"
+#include <iostream>
+
+int main(int argc, char* argv[]) {
+   std::string model_path = argv[1];
+   ov::genai::LLMPipeline pipe(model_path, "CPU");
+   std::cout << pipe.generate("The Sun is yellow because", ov::genai::max_new_tokens(100));
+}
+```
+
+### Sample notebooks using this API
+
+(TBD)
+
+</details>
+
```

Comment:
should we also point out to a list of supported models? https://github.com/openvinotoolkit/openvino.genai/blob/master/src/docs/SUPPORTED_MODELS.md#large-language-models